### PR TITLE
WAS16 cleanup client data

### PR DESCRIPTION
**Pull request by was4**
clnt_dg_call with clnt_req
(rpc_dplx_internal) Consolidate common init and free
(clnt_internal) time_not_ok
Wrap cx_data
Remove unusable CLSET_CONNECT
Remove unusable CLSET_ASYNC

### CODE REVIEW

File: src/clnt_rdma.c [hunk 1]
```diff
@@ -101,13 +129,10 @@ clnt_rdma_create(RDMAXPRT *xprt,		/* init but NOT connect()ed descriptor */
 	 */
 //	u_int sendsz = 8*1024;
 //	u_int recvsz = 4*8*1024;
```
Comment (dang) [inline src/clnt_rdma.c]: The commented versions should probably go too.

**Review by dang — COMMENTED** [review]
Looks good, one minor nit.

**Comment by was4** [comment]
On 10/9/17 11:29 AM, Daniel Gryniewicz wrote:
> *@dang* commented on this pull request.
> 
> Looks good, one minor nit.
> 
Done, and a bit more cleanup.  Still more to do,
but this should be a good spot to stop.

File: src/clnt_internal.h [hunk 2]
```diff
@@ -65,4 +65,16 @@ clnt_data_destroy(struct cx_data *cx)
...
 }
 
+
+/*
+ * Make sure that the time is not garbage.  0 or - value is *NOT* allowed.
+ */
+static inline bool
+time_not_ok(struct timeval *t)
+{
+	return (t->tv_sec < 0 || t->tv_sec > 99 ||
```
Comment (dang) [inline src/clnt_internal.h]: I disagree with this.  The caller should be able to pick a large timeout, if they want.  99 it much to arbitrarily small.  For example, the TCP timeout is 60 minutes.  Leave it large.

**Comment by was4** [comment]
> I disagree with this. The caller should be able to pick a large timeout, if they want. 99 it much to arbitrarily small. For example, the TCP timeout is 60 minutes. Leave it large.
> 
No, 99 is not small enough.  I used it because there were existing 60
second timers.  This is specifically for retransmission timeouts.

BTW, there's no such thing as a "TCP timeout" of "60 minutes".
Explicitly forbidden. Nobody would allow such a retransmission timeout.

**Comment by dang** [comment]
ntirpc is a library.  It is up to the application to decide things like timeouts, as long as they're not baked into the protocol.  Fix the timeouts where they're used, don't limit them in the library.

**Comment by was4** [comment]
On 10/12/17 11:47 AM, Daniel Gryniewicz wrote:
> ntirpc is a library. It is up to the application to decide things like timeouts, as long as they're not baked into the protocol. Fix the timeouts where they're used, don't limit them in the library.
> 
We discussed on the community call today, and this can go in as-is.

Library calls should always check user parameters for valid values.

In a future patch, after I've found and fixed all the longest timeouts,
this will be dropped further to 10 seconds.  3+ round trips to the moon.

**Comment by was4** [comment]
On 10/17/17 12:20 PM, William Allen Simpson wrote:
> Library calls should always check user parameters for valid values.
> 
The compromised proposition is to use warnx _WARN for too large parameters.


> In a future patch, after I've found and fixed all the longest timeouts,
> this will be dropped further to 10 seconds.  3+ round trips to the moon.

The warnx will allow finding and fixing all those > 10 seconds.

**Review by dang — APPROVED** [review]

**Pull request merged**